src/lib/mongoBackend/mongoUpdateContextAvailabilitySubscription.cpp

```diff
@@ -155,38 +168,71 @@ HttpStatusCode mongoUpdateContextAvailabilitySubscription
   int count = sub.hasField(CASUB_COUNT) ? getIntFieldF(sub, CASUB_COUNT) : 0;
 
   /* The hasField check is needed due to lastNotification/count could not be present in the original doc */
-  if (sub.hasField(CASUB_LASTNOTIFICATION)) {
-      newSub.append(CASUB_LASTNOTIFICATION, getIntFieldF(sub, CASUB_LASTNOTIFICATION));
+  if (sub.hasField(CASUB_LASTNOTIFICATION))
+  {
+    newSub.append(CASUB_LASTNOTIFICATION, getIntFieldF(sub, CASUB_LASTNOTIFICATION));
   }
-  if (sub.hasField(CASUB_COUNT)) {
-      newSub.append(CASUB_COUNT, count);
+
+  if (sub.hasField(CASUB_COUNT))
+  {
+    newSub.append(CASUB_COUNT, count);
   }
 
-  // FIXME P5: RenderFormat right now hardcoded to "JSON" (NGSI_V1_LEGACY), in the future the RenderFormat will be taken from the payload
+  //
+  // FIXME P5: RenderFormat right now hardcoded to "JSON" (NGSI_V1_LEGACY),
+  //           in the future the RenderFormat will be taken from the payload
+  //
+
   /* Adding format to use in notifications */
   newSub.append(CASUB_FORMAT, "JSON");
 
   /* Update document in MongoDB */
-  if (!collectionUpdate(getSubscribeContextAvailabilityCollectionName(tenant), BSON("_id" << OID(requestP->subscriptionId.get())), newSub.obj(), false, &err))
+
+  //
+  // FIXME P4: Strange errors in two unit tests:
```

Comment:
Maybe the same solution you used for https://github.com/telefonicaid/fiware-orion/pull/2909/files#r118671472 also solves this case?